### ISSUE TITLE
About training details

### ISSUE THREAD

**Issue by DYF-AI** [preddy5/Im2Vec#5]
Dear author:
Regarding the training details, I have the following questions?
1. How many image data are used to train emoji? I saw only 13 images in data/emoji/train and logs in the training. Will you also train with these data?
2. Is the training process completely end-to-end? And do I need to set other configurations in the yaml file in config in order to train with better results? Did you follow the training in the yaml file?
3. I run CUDA_VISIBLE_DEVICES=1 python run.py -c configs/emoji.yaml， and I have trained 2100 epochs so far, and the loss does not seem to decrease practically.
```
Epoch 02177: val_loss  was not in top 1
Epoch 2178: 100%|█| 13/13 [00:11<00:00,  1.10it/s, loss=3.677, v_num=110, Reconstruction_Loss=3.67, KLD=0, aux_loss=0, other losses=0beta:  0.0
learning rate:  0.0001

Epoch 02178: val_loss  was not in top 1
Epoch 2179: 100%|█| 13/13 [00:13<00:00,  1.03s/it, loss=3.731, v_num=110, Reconstruction_Loss=3.59, KLD=0, aux_loss=0, other losses=0beta:  0.0
learning rate:  0.0001

Epoch 02179: val_loss  was not in top 1
Epoch 2180: 100%|█| 13/13 [00:13<00:00,  1.01s/it, loss=3.689, v_num=110, Reconstruction_Loss=3.56, KLD=0, aux_loss=0, other losses=0beta:  0.0
learning rate:  0.0001

Epoch 02180: val_loss  was not in top 1
Epoch 2181: 100%|█| 13/13 [00:12<00:00,  1.00it/s, loss=3.693, v_num=110, Reconstruction_Loss=3.33, KLD=0, aux_loss=0, other losses=0beta:  0.0
learning rate:  0.0001
```
![image](https://user-images.githubusercontent.com/54470509/120576905-18f6ef80-c456-11eb-9129-c2e191a9a7c0.png)
reconstruction 
![image](https://user-images.githubusercontent.com/54470509/120576956-2d3aec80-c456-11eb-9980-14b24604fc27.png)

Is this result normal? Is this due to the small training data set? Or yaml file configuration?

**Comment by preddy5** [comment]
Hey @DYF-AI 
i) Yes there are only a few images in the emojis dataset. 
ii) Yes the training process is end-to-end. Could try training by setting https://github.com/preddy5/Im2Vec/blob/master/configs/emoji.yaml#L13 to 0
iii) Also it looks like you only have 2 curves can you recheck if https://github.com/preddy5/Im2Vec/blob/master/models/vector_vae_nlayers.py#L44 has 4 colors and 4 colors correspond to the ones in the dataset. No the results should look better. Refer to https://github.com/preddy5/Im2Vec/tree/master/logs/VectorVAEnLayers/version_110 for an example training run.

**Comment by DYF-AI** [comment]
Thank you very much for your reply, your suggestion is very effective：
ii) Yes the training process is end-to-end. Could try training by setting https://github.com/preddy5/Im2Vec/blob/master/configs/emoji.yaml#L13 to 0
![image](https://user-images.githubusercontent.com/54470509/120948705-684e6000-c775-11eb-8d1f-fa4b1420beef.png)
This is the result of 185 epochs, which is much better than before，thank...

**Comment by DYF-AI** [comment]
Hey, Dear Author
Epoch 401: 100%|██████████████| 13/13 [00:07<00:00,  1.83it/s, loss=0.170, v_num=110, Reconstruction_Loss=0.0777, KLD=0, aux_loss=0, other losses=0]beta:  0.0
learning rate:  0.00029936846961918924

I have another question. I saw KLD=0, aux_loss=0, other losses=0 during training. Do these losses need to be turned on during training? In other words, will the result be better after turning it on?

**Comment by preddy5** [comment]
Hey @DYF-AI 
KLD is the KL divergence loss like in a VAE
aux_loss is for the auxiliary network we discussed in the main paper. 
other losses is the loss term to encourage minimal surface area for the curves.

using other_losses term after 500 epoch will improve the results a tiny bit, KLD and aux should not affect your results.

**Comment by DYF-AI** [comment]
Okay, thank you very much for your reply, this is a very interesting and meaningful project, thanks. I want to use path-to-path to generate fonts based on this project...

**Comment by preddy5** [comment]
@DYF-AI you are welcome. All the best with the project :)
